app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js

```diff
@@ -158,22 +126,47 @@ class TableVisualization extends Visualization {
                 },
             ])
             this.agGridOptions.api.setRowData([{ Error: parsedData.error }])
-        } else if (parsedData.json != null && isMatrix(parsedData.json)) {
-            columnDefs = parsedData.json[0].map((_, i) => ({ field: i.toString() }))
-            rowData = parsedData.json
+        } else if (parsedData.type === 'Matrix') {
+            let defs = [{ field: '#' }]
+            for (let i = 0; i < parsedData.column_count; i++) {
+                defs.push({ field: i.toString() })
+            }
+            columnDefs = defs
+            rowData = addRowIndex(parsedData.json)
+            dataTruncated = parsedData.all_rows_count !== parsedData.json.length
+        } else if (parsedData.type === 'Object_Matrix') {
```

Comment:
Why are the cases `isObjectMatrix(...)` and `parsedData.type == 'Object_Matrix'` distinguished? They look very similar. I think it would be good to add some comment explaining what is the difference between these cases, as it may be very confusing - reading this code I see the "same" case handled twice in a slightly different way and I just don't know why.

Reply:
Ah, I see. Makes sense - can we have a comment in the code? Preferably with an explanation when can we remove this legacy support? (I assume we don't want to keep it forever)